selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/php/PhpAssistantFeaturesTest.java

```diff
@@ -0,0 +1,186 @@
+  @Inject private AssistantFindPanel assistantFindPanel;
+
+  @BeforeClass
+  public void setup() throws Exception {
+    testProjectServiceClient.importProject(
+        ws.getId(), Paths.get(RESOURCE.toURI()), PROJECT, ProjectTemplates.PHP);
+
+    // open IDE
+    ide.open(ws);
+    loader.waitOnClosed();
+    projectExplorer.waitItem(PROJECT);
+    notificationPopup.waitProgressPopupPanelClose();
+
+    // open project tree
+    projectExplorer.quickExpandWithJavaScript();
+    projectExplorer.openItemByPath(PATH_TO_INDEX_PHP);
+    editor.waitActive();
+  }
+
+  @Test
+  public void codeShouldBeCommented() {
+    projectExplorer.openItemByPath(PATH_TO_INDEX_PHP);
+    editor.waitActive();
+    editor.waitTextIntoEditor(EXPECTED_ORIGINAL_TEXT);
+
+    editor.setCursorToLine(4);
+    performCommentAction();
+    editor.waitTextIntoEditor(EXPECTED_COMMENTED_TEXT);
+
+    performCommentAction();
+    editor.waitTextIntoEditor(EXPECTED_ORIGINAL_TEXT);
+  }
+
+  @Test
+  public void hoverShouldBeDisplayedWithExpectedText() {
+    projectExplorer.openItemByPath(PATH_TO_INDEX_PHP);
+    editor.waitActive();
+
+    editor.moveCursorToText(TEXT_FOR_HOVERING);
+    editor.waitTextInHoverPopup(EXPECTED_HOVER_POPUP_TEXT);
+  }
+
+  @Test
+  public void findReferenceNodeShouldBeDisplayed() {
+    projectExplorer.openItemByPath(PATH_TO_INDEX_PHP);
+    editor.waitActive();
+
+    editor.goToCursorPositionVisible(15, 8);
+    menu.runCommand(ASSISTANT, FIND_REFERENCES);
+    findReferencesConsoleTab.waitReferenceWithText(EXPECTED_REFERENCE_TEXT);
+  }
+
+  @Test
+  public void signatureShouldBeDisplayed() {
+    projectExplorer.openItemByPath(PATH_TO_INDEX_PHP);
+    editor.waitActive();
+
+    editor.goToCursorPositionVisible(15, 22);
+    editor.typeTextIntoEditor(ENTER.toString());
+    editor.waitCursorPosition(16, 1);
+    editor.typeTextIntoEditor(TEXT_FOR_HOVERING + "(");
+    editor.waitTextIntoEditor(EXPECTED_TEXT_AFTER_TYPING);
+    waitExpectedTextIntoShowHintsPopup(EXPECTED_HINT_TEXT);
+    editor.deleteCurrentLine();
+  }
+
+  @Test
+  public void goToSymbolFeatureShouldWork() {
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.openItemByPath(PROJECT + "/" + LIB_TAB_NAME);
+    editor.waitActive();
+
+    editor.waitTabIsPresent(LIB_TAB_NAME);
+    menu.runCommand(ASSISTANT, GO_TO_SYMBOL);
+    assistantFindPanel.waitActionNodeContainsText(EXPECTED_GO_TO_SYMBOL_TEXT);
+  }
+
+  @Test
+  public void checkEditor() {
+    projectExplorer.openItemByPath(PATH_TO_INDEX_PHP);
+    editor.waitActive();
+    editor.selectTabByName(INDEX_TAB_NAME);
+    editor.waitActiveTabFileName(INDEX_TAB_NAME);
+    editor.waitActive();
+
+    menu.runCommand(ASSISTANT, FIND_PROJECT_SYMBOL);
+    assistantFindPanel.waitForm();
+    assistantFindPanel.typeToInputField(FIND_PROJECT_SEARCHING_TEXT);
+    assistantFindPanel.waitActionNodeContainsText(EXPECTED_FIND_PROJECT_TEXT);
+  }
+
+  private void performCommentAction() {
```

Comment:
It could be useful for another LS tests. What about moving it to CodenvyEditor class and rename to "lineComment()"?